arangod/Cluster/Utils/ShardID.h

```diff
@@ -0,0 +1,161 @@
+  ShardID(ShardID const&) = default;
+  ShardID(ShardID&&) = default;
+  ShardID& operator=(ShardID const&) = default;
+  ShardID& operator=(ShardID&&) = default;
+
+  operator std::string() const {
+    return "s" + std::to_string(id);
+  }
+
+  operator arangodb::velocypack::Value() const {
+    return arangodb::velocypack::Value("s" + std::to_string(id));
+  }
+
+  friend auto operator<=>(ShardID const&, ShardID const&) = default;
+  friend bool operator==(ShardID const&, ShardID const&) = default;
+
+  bool operator==(std::string_view other) const {
+    return other == std::string{*this};
+  }
+
+  bool operator==(std::string const& other) const {
+    return other == std::string{*this};
+  }
+
+  bool isValid() const noexcept {
+    // We can never have ShardID 0. So we use it as invalid value.
+    return id != 0;
+  }
+
+  // Add an inspector implementation, shardIDs will be serialized and deserialized as "s" + number
+  // for compatibility reasons.
+  template<class Inspector>
+  inline friend auto inspect(Inspector& f, ShardID& x) -> arangodb::inspection::Status {
+    if constexpr (Inspector::isLoading) {
+      std::string v;
+      auto res = f.apply(v);
+      if (res.ok()) {
+        auto result = shardIdFromString(v);
+        if (result.fail()) {
+          // NOTE: For some reason we cannot inline the errorMessage() into the return
+          // in that case the string_view variant will be selected, which is no valid input
+          // for status.
+          std::string msg{std::move(result.errorMessage())};
+          return {std::move(msg)};
+        }
+        x = result.get();
+        return {};
+      }
+      return res;
+    } else {
+      return f.apply("s" + std::to_string(x.id));
+    }
+  }
+
+  uint64_t id;
+};
+
+// Make ShardID logable
+static inline std::ostream& operator<<(std::ostream& o, arangodb::ShardID const& r) {
+  o << "s" << r.id;
+  return o;
+}
+
+}  // namespace arangodb
+
+// Make ShardID ready to be used as key in std::unordered_map
+template<>
+struct std::hash<arangodb::ShardID> {
+  [[nodiscard]] auto operator()(arangodb::ShardID const& v) const noexcept
+      -> std::size_t {
+    return std::hash<uint64_t>{}(v.id);
+  }
+};
+
+// Make ShardID fmt::formatable
+template<>
+struct fmt::formatter<arangodb::ShardID> {
+  template <typename ParseContext>
+  constexpr auto parse(ParseContext& ctx) { return ctx.begin(); }
+
+  template<typename FormatContext>
+  auto format(arangodb::ShardID const& shardId, FormatContext& ctx) {
+    return format_to(ctx.out(), "{}", "s" + std::to_string(shardId.id));
+  }
+};
+
+// Allow ShardID to be added to std::strings
+static inline std::string operator+(std::string const& text, arangodb::ShardID const& s) {
+  return text + "s" + std::to_string(s.id);
+}
+// Allow ShardID to be added to std::strings
+static inline std::string operator+(arangodb::ShardID const& s, std::string const& text) {
+  return  "s" + std::to_string(s.id) + text;
+}
```

Comment:
Instead of declaring these static inline, we should move the implementation into a .cpp file.
```suggestion
static inline std::string operator+(std::string const& text, arangodb::ShardID const& s) {
  return absl::StrCat(text, "s", s.id);
}
// Allow ShardID to be added to std::strings
static inline std::string operator+(arangodb::ShardID const& s, std::string const& text) {
  return absl::StrCat("s", s.id, text);
}
```

Reply:
Done